File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

def dfs(v, visited, graph, result):
    visited[v] = True
    result.append(v)
    for i in range(1, len(graph)):
        if not visited[i] and graph[v][i] == 1:
            dfs(i, visited, graph, result)

def bfs(v, visited, graph, result):
    queue = deque([v])
    visited[v] = True
    while queue:
        node = queue.popleft()
        result.append(node)
        for i in range(1, len(graph)):
            if not visited[i] and graph[node][i] == 1:
                queue.append(i)
                visited[i] = True

def main():
    N, M, V = map(int, input().split())  
    graph = [[0] * (N + 1) for _ in range(N + 1)]  

    for _ in range(M):
        u, v = map(int, input().split())
        graph[u][v] = graph[v][u] = 1  

    visited_dfs = [False] * (N + 1)
    visited_bfs = [False] * (N + 1)
    
    dfs_result = []
    bfs_result = []
    
    dfs(V, visited_dfs, graph, dfs_result)
    
    bfs(V, visited_bfs, graph, bfs_result)
    
    print(" ".join(map(str, dfs_result)))
    print(" ".join(map(str, bfs_result)))


if __name__ == "__main__":
    main()

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const input = require("fs").readFileSync("/dev/stdin").toString().split('\n');
// const input = require("fs").readFileSync('./input.txt').toString().split('\n');
const [N, M, V] = input[0].split(" ").map(Number);
let graph = [];
let result = '';

function dfs(v, visited) {
  if (visited[v]) {
    return;
  }
  result += v + ' ';
  visited[v] = true;
  for (let i = 1; i <= N; i++) {
    if (!visited[i] && graph[v][i] === 1) {
      dfs(i, visited);
    }
  }
}

function bfs(v, visited) {
  let queue = [v];
  visited[v] = true;
  while (queue.length > 0) {
    v = queue.shift();
    result += v + ' ';
    for (let i = 1; i <= N; i++) {
      if (!visited[i] && graph[v][i] === 1) {
        queue.push(i);
        visited[i] = true;
      }
    }
  }
}

for (let i = 0; i <= N; i++) {
  graph[i] = Array(N + 1).fill(0);
}

for (let i = 1; i <= M; i++) {
  const [x, y] = input[i].split(" ").map(Number);
  graph[x][y] = graph[y][x] = 1;
}

dfs(V, Array(N + 1).fill(false));
result += '\n';
bfs(V, Array(N + 1).fill(false));

console.log(result);
